Userland/Libraries/LibC/time.cpp

```diff
@@ -180,7 +186,14 @@ struct tm* gmtime_r(const time_t* t, struct tm* tm)
 char* asctime(const struct tm* tm)
 {
     static char buffer[69];
-    strftime(buffer, sizeof buffer, "%a %b %e %T %Y", tm);
+    return asctime_r(tm, buffer);
+}
+
+char* asctime_r(const struct tm* tm, char* buffer)
+{
+    // Spec states buffer must be at least 26 bytes.
+    constexpr size_t assumed_len = 26;
+    strftime(buffer, assumed_len, "%a %b %e %T %Y", tm);
```

Comment:
This looked suspicious, but it's indeed just a crappy API. I checked glibc and musl, and they have these comments which I enjoyed :smile:

https://github.com/bminor/glibc/blob/81f6dd2135ea761832965bc7518e2ddf949480af/time/asctime.c#L42-L48
https://github.com/ifduyue/musl/blob/cfdfd5ea3ce14c6abf7fb22a531f3d99518b5a1b/src/time/asctime_r.c#L16-L22

It seems this is missing a trailing newline though?

Reply:
Nice spot! Yea looks like that was missing from the original implementation.

I added a commit which fixes this (crediting you), and updated the tests to verify it's presence.